src/vivarium/framework/resource/group.py

```diff
@@ -71,14 +83,26 @@ def initializer(self) -> Callable[[SimulantData], None]:
     @property
     def dependencies(self) -> list[str]:
         """The long names (including type) of dependencies for this group."""
-        return [dependency.resource_id for dependency in self._dependencies]
+        if not all(isinstance(dep, Resource) for dep in self._dependencies):
+            raise ResourceError(
+                "Resource group has not been finalized; dependencies are still strings."
```

Comment:
Add information about which dependencies are strings and which resource group this is.